Talked in DM with @haampie Currently:
```console
$ spack install libbsd
==> Installing libbsd-0.11.3-gxm35f3x2ht67znxyy24ih64srydcsgo
==> No binary for libbsd-0.11.3-gxm35f3x2ht67znxyy24ih64srydcsgo found: installing from source
==> Fetching https://libbsd.freedesktop.org/releases/libbsd-0.11.3.tar.xz
########################################################################################################################################### 100.0%
==> Ran patch() for libbsd
==> libbsd: Executing phase: 'autoreconf'
==> libbsd: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/tmp/culpo/spack-stage/spack-stage-libbsd-0.11.3-gxm35f3x2ht67znxyy24ih64srydcsgo/spack-src/configure' '--prefix=/home/culpo/PycharmProjects/spack/opt/spack/linux-ubuntu18.04-broadwell/gcc-10.1.0/libbsd-0.11.3-gxm35f3x2ht67znxyy24ih64srydcsgo'

1 error found in build log:
     91    checking whether /home/culpo/PycharmProjects/spack/lib/spack/env/gcc/gcc understands -c and -o together... (cached) yes
     92    checking dependency style of /home/culpo/PycharmProjects/spack/lib/spack/env/gcc/gcc... (cached) gcc3
     93    checking whether ln -s works... yes
     94    checking for unif01_CreateExternGenBits in -ltestu01... no
     95    checking for library containing dlsym... -ldl
     96    checking for library containing MD5Update... no
  >> 97    configure: error: cannot find required MD5 functions in libc or libmd

See build log for details:
  /tmp/culpo/spack-stage/spack-stage-libbsd-0.11.3-gxm35f3x2ht67znxyy24ih64srydcsgo/spack-build-out.txt
```
This needs to be investigated before merging the PR.

--- a/var/spack/repos/builtin/packages/libbsd/package.py
+++ b/var/spack/repos/builtin/packages/libbsd/package.py
@@ -36,6 +36,8 @@ class Libbsd(AutotoolsPackage):
     # https://gitlab.freedesktop.org/libbsd/libbsd/issues/1
     conflicts('platform=darwin')
 
+    depends_on('libmd', when='@0.11:')
+
     def patch(self):
         # Remove flags not recognized by the NVIDIA compiler
         if self.spec.satisfies('%pgi') or self.spec.satisfies('%nvhpc'):

--- a/var/spack/repos/builtin/packages/libmd/package.py
+++ b/var/spack/repos/builtin/packages/libmd/package.py
@@ -1,0 +1,22 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Libmd(AutotoolsPackage):
+    """This library provides message digest functions found on BSD systems
+       either on their libc (NetBSD, OpenBSD) or libmd (FreeBSD, DragonflyBSD,
+       macOS, Solaris) libraries and lacking on others like GNU systems."""
+
+    homepage = "https://www.hadrons.org/software/libmd/"
+    url      = "https://archive.hadrons.org/software/libmd/libmd-1.0.3.tar.xz"
+
+    maintainers = ['haampie']
+
+    version('1.0.3', sha256='5a02097f95cc250a3f1001865e4dbba5f1d15554120f95693c0541923c52af4a')
+    version('1.0.2', sha256='dc66b8278f82e7e1bf774fbd4bc83a0348e8f27afa185b2c2779cfcb3da25013')
+    version('1.0.1', sha256='e14eeb931cf85330f95ff822262d3033125488dfb2f867441e36e2d2c4a34c71')
+    version('1.0.0', sha256='f21aea69f6411cb4307cda1f6378c7ed07830202b5f4cb9e64f681fdaf2d64c7')

